prost-types/src/lib.rs

```diff
@@ -33,6 +37,39 @@ pub use protobuf::*;
 const NANOS_PER_SECOND: i32 = 1_000_000_000;
 const NANOS_MAX: i32 = NANOS_PER_SECOND - 1;
 
+impl Any {
+    /// Serialize the given message type `M` as [`Any`].
+    pub fn from_msg<M>(msg: &M) -> Result<Self, EncodeError>
+    where
+        M: Name,
+    {
+        let type_url = M::type_url();
+        let mut value = Vec::new();
+        Message::encode(msg, &mut value)?;
+        Ok(Any { type_url, value })
+    }
+
+    /// Decode the given message type `M` from [`Any`], validating that it has
+    /// the expected type URL.
+    pub fn to_msg<M>(&self) -> Result<M, DecodeError>
+    where
+        M: Default + Name + Sized,
+    {
+        let expected_type_url = M::type_url();
+
+        if self.type_url == expected_type_url {
```

Comment:
This check is naive. Apparently we should split at "/" and compare the fullnames afterward to see if they match

Reply:
I added more robust type URL matching in [01fde2d](https://github.com/tokio-rs/prost/pull/896/commits/01fde2daebb7e75d5376a560ee50ac900119890c)